vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -54,8 +53,9 @@ def assign_account
         occupancy.update(account: accounts.first)
       end
 
-      def user_can_purchase_secure_room?
-        occupancy.user.accounts_for_product(occupancy.secure_room).present?
+      def user_exempt_from_purchase?
+        in_reader = occupancy.secure_room.card_readers.ingress.first
```

Comment:
Can you add some commentary on why this works? I'm pretty sure it's fine, but it's definitely not intuitive.